src/services/SolanaService.ts

```diff
@@ -0,0 +1,250 @@
+import { 
+  getDomainKey, 
+  NameRegistryState, 
+  Record, 
+  getIpfsRecord, 
+  performReverseLookup,
+  getAllDomains,
+  resolve
+} from "@bonfida/spl-name-service";
+import { AllDIDErrorCode } from '../errors/AllDIDError';
+
+export function createProvider(
+  endpoint: string,
+  commitmentOrConfig?: Commitment | ConnectionConfig
+): Connection {
+  return new Connection(endpoint, commitmentOrConfig)
+}
+
+export type SolanaServiceOptions = {
+  provider: Connection,
+  network: string
+}
+
+function getDwebKeys () {
+  return [
+    "IPFS",
+    "ARWV",
+    "SHDW",
+    "Url"
+  ]
+}
+
+function getTextKeys () {
+  return [
+    'POINT',
+    'Injective',
+    'Pic'
+  ]
+}
+
+function getProfileKeys () {
+  return [
+    "Email",
+    "Discord",
+    "Github",
+    "Reddit",
+    "Twitter",
+    "Telegram",
+  ]
+}
+
+function getAddressKeys () {
+  return [
+    'ETH',
+    'BTC',
+    'SOL',
+    'LTC',
+    'DOGE',
+    // 'BSC' // @bonfida/spl-name-service v0.1.51 does not support BSC
+  ]
+}
+
+enum KeyPrefix {
+  Address= 'address',
+  Profile= 'profile',
+  Dweb = 'dweb',
+  Text = 'text'
+} 
+
+
+export class SolanaService extends NamingService {
+  serviceName = 'solana'
+  
+  provider: Connection
+  constructor (options: SolanaServiceOptions) {
+    super()
+    this.provider = options.provider
+  }
+
+  isSupported (name: string): boolean {
+    return /^.+\.sol$/.test(name)
+  }
+
+  protected async isValidPubkey(pubkey: PublicKey): Promise<boolean> {
+    const nameAccount = await this.provider.getAccountInfo(pubkey)
+    return nameAccount ? true : false
+  }
+
+  protected getSolAddress (data: Buffer ): string {
+    return new PublicKey(data.slice(0, 32)).toString()
+  }
+
+  async isRegistered (name: string): Promise<boolean> {
+    const { pubkey } = await getDomainKey(name);
+    return this.isValidPubkey(pubkey)
+  }
+
+  async isAvailable (name: string): Promise<boolean> {
+    return !(await this.isRegistered(name))
+  }
+
+  async owner (name: string): Promise<string> {
+    const { pubkey } = await getDomainKey(name);
+    const isValid = await this.isValidPubkey(pubkey)
+    if (!isValid) return null
+    const { registry, nftOwner } = await NameRegistryState.retrieve(
+      this.provider,
+      pubkey
+    );
+    return nftOwner ? nftOwner.toString() : registry.owner.toString()
+  }
+
+  manager (name: string): Promise<string> {
+    return this.owner(name)
+  }
+
+  tokenId (name: string): Promise<string> {
+    this.throwError('Unsupported Method', AllDIDErrorCode.UnsupportedMethod)
+    return null
+  }
+
+  protected makeRecordItem (key: string): RecordItem {
+    const keyArray = key.split('.')
+    const type = keyArray.length > 1 ? keyArray[0] : ''
+    const subtype = keyArray[keyArray.length - 1]
+    return {
+      key,
+      type,
+      subtype,
+      label: '',
+      value: null,
+      ttl: 0,
+    }
+  }
+
+  protected makeRecordKey (subtype: string): string {
```

Comment:
For long-term maintenance, it would be better to add some comments, indicating the input and output formats.